src/user.ts

```diff
@@ -0,0 +1,232 @@
+type UserProperties = { [key: string]: unknown }
+
+export class User {
```

Comment:
This name is going to overlap with the User type which might get confusing. We can't change that for back-compat, but maybe this could be UserObject or UserHelper... naming is hard.

Reply:
Updated to UserClass. How is that?